### PR TITLE
Cleanup after migrating UI to `OfflinePlugin`

### PR DESCRIPTION
**Pull request by MarinTolic**
### 🎯 Goal

Our efforts to Migrate our UI modules over to OP require some minor cleanup effort.

### 🛠 Implementation details

- Remove unnecessary OptIns now that `OfflinePlugin` and the related structure is no longer marked with `@InternalStreamChatApi`
- Remove unnecessary message preloading while requesting `ChannelState` where there is no need for it.

### 🎨 ~~UI Changes~~

**No UI Changes**

### 🧪 Testing

Test basic functionalities in UI components such as 

- Checking that typing in a message produces a typing update on another device
- Send a message
- Check that `MessageListHeaderView` is behaving as expected (Online status, typing updates, the correct number of members, etc)

#### General
- [ ] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [ ] ~~PR is linked to the GitHub issue it resolves~~

#### Code & documentation
- [ ] ~~Changelog is updated with client-facing changes~~
- [ ] ~~New code is covered by unit tests~~
- [ ] ~~Comparison screenshots added for visual changes~~
- [ ] ~~Affected documentation updated (KDocs, docusaurus, tutorial)~~

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] ~~UI Changes correct (before & after images)~~
- [ ] ~~Bugs validated (bugfixes)~~
- [ ] New feature tested and works
- [ ] ~~Release notes and docs clearly describe changes~~
- [ ] ~~All code we touched has new or updated KDocs~~

### 🎉 GIF

cleanup

### CODE REVIEW

File: stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/viewmodel/MessageInputViewModel.kt [hunk 1]
```diff
@@ -46,7 +45,7 @@ public class MessageInputViewModel @JvmOverloads constructor(
      * Holds information about the current channel and is actively updated.
      */
     public val channelState: ChannelState =
-        chatClient.watchChannelAsState(cid, MessageListViewModel.DEFAULT_MESSAGES_LIMIT, viewModelScope)
+        chatClient.watchChannelAsState(cid, 0, viewModelScope)
```
Comment (adasiewiczr) [inline stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/viewmodel/MessageInputViewModel.kt:48]: Can we use named parameters here or extract 0 to a constant? It might be hard to guess what does this magic number mean

**Review by adasiewiczr — COMMENTED** [review]
I left one comment but besides that looks good to me!

### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [ ] ~~UI Changes correct (before & after images)~~
- [ ] ~~Bugs validated (bugfixes)~~
- [x] New feature tested and works
- [ ] ~~Release notes and docs clearly describe changes~~
- [ ] ~~All code we touched has new or updated KDocs~~

File: stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/viewmodel/messages/MessageListViewModel.kt [hunk 2]
```diff
@@ -50,11 +50,11 @@ import io.getstream.chat.android.livedata.utils.Event as EventWrapper
  * @param globalState Global state of OfflinePlugin. Contains information
  * such as the current user, connection state, unread counts etc.
  */
-public class MessageListViewModel @JvmOverloads constructor(
+public class MessageListViewModel constructor(
```
Comment (filbabic) [inline stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/viewmodel/messages/MessageListViewModel.kt:53]: we can also remove the `constructor` keyword given that we no longer have annotations

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/actions/internal/ChannelActionsViewModel.kt [hunk 3]
```diff
@@ -31,7 +31,7 @@ internal class ChannelActionsViewModel(
      * Holds information about the current channel and is actively updated.
      */
     val channelState: ChannelState =
-        chatClient.watchChannelAsState(cid, DEFAULT_MESSAGE_LIMIT, viewModelScope)
+        chatClient.watchChannelAsState(cid, 0, viewModelScope)
```
Comment (filbabic) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/actions/internal/ChannelActionsViewModel.kt:34]: same as above - I'd love a constant or a named argument

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/header/viewmodel/MessageListHeaderViewModel.kt [hunk 4]
```diff
@@ -1,15 +1,96 @@
...
     chatClient: ChatClient = ChatClient.instance(),
-) : BaseMessageListHeaderViewModel(cid, chatClient)
+    globalState: GlobalState = chatClient.globalState,
+) : ViewModel() {
+
+    /**
+     * Holds information about the current channel and is actively updated.
+     */
+    public val channelState: ChannelState =
+        chatClient.watchChannelAsState(cid, 0, viewModelScope)
```
Comment (filbabic) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/header/viewmodel/MessageListHeaderViewModel.kt:37]: .

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/typing/viewmodel/TypingIndicatorViewModel.kt [hunk 5]
```diff
@@ -35,12 +35,4 @@ public class TypingIndicatorViewModel(
         channelState.typing.map { typingEvent ->
             typingEvent.users
         }.asLiveData()
-
-    private companion object {
-
-        /**
-         * The default limit for messages count in requests.
-         */
-        private const val DEFAULT_MESSAGE_LIMIT: Int = 30
```
Comment (filbabic) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/typing/viewmodel/TypingIndicatorViewModel.kt:44]: Also - most of these functions used this as the default limit and now it's `0`, any reason why this change?

**Review by filbabic — COMMENTED** [review]
Added some comments!

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/typing/viewmodel/TypingIndicatorViewModel.kt (hunk 5 above)
Comment (MarinTolic) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/typing/viewmodel/TypingIndicatorViewModel.kt:44]: The reason why it's a 0 is because there is no benefit of message preloading here, it's just unnecessary because other than `MessageListViewViewModel`, our VMs do not benefit from having messages stored.

The reason why it's not just left empty is because @adasiewiczr  removed the default argument particularly for this reason of people leaving it mistakenly empty and pre loading messages when they should not be doing so.

**Review by adasiewiczr — APPROVED** [review]
LGTM
